UnboundLib/Patches/GoBack.cs

```diff
@@ -1,54 +1,54 @@
-                {
-                    ToggleLevelMenuHandler.instance.mapMenuCanvas.SetActive (false);
-                    return false;
-                }
-                
-                if (!ToggleCardsMenuHandler.disableEscapeButton && ToggleCardsMenuHandler.cardMenuCanvas.activeInHierarchy)
-                {
-                    ToggleCardsMenuHandler.SetActive(ToggleCardsMenuHandler.cardMenuCanvas.transform, false);
-                    if(ToggleCardsMenuHandler.menuOpenFromOutside) ToggleCardsMenuHandler.Close();
-                    return false;
-                }
-            }
-
-            if (ModOptions.inPauseMenu)
-            {
-                return false;
-            }
-            
-
-            return true;
-        }
-    }
-}
\ No newline at end of file
+﻿// using HarmonyLib;
+// using UnboundLib.Utils.UI;
+// using UnityEngine;
+//
+// namespace UnboundLib.Patches
+// {
+//     [HarmonyPatch(typeof(GoBack))]
+//     public class GoBackPatch
+//     {
+//         [HarmonyPatch("Update")]
+//         [HarmonyPrefix]
+//         private static bool Update()
+//         {
+//             if (Input.GetKeyDown(KeyCode.Escape))
+//             {
+//                 if (ToggleLevelMenuHandler.instance.mapMenuCanvas.transform.Find("MapMenu/InfoMenu").gameObject.activeInHierarchy)
+//                 {
+//                     ToggleLevelMenuHandler.instance.mapMenuCanvas.transform.Find("MapMenu/InfoMenu").gameObject
+//                         .SetActive(false);
+//                     return false;
+//                 }
+//                 
+//                 if (ToggleCardsMenuHandler.cardMenuCanvas.transform.Find("CardMenu/InfoMenu").gameObject.activeInHierarchy)
+//                 {
+//                     ToggleCardsMenuHandler.cardMenuCanvas.transform.Find("CardMenu/InfoMenu").gameObject
+//                         .SetActive(false);
+//                     if(ToggleCardsMenuHandler.menuOpenFromOutside) ToggleCardsMenuHandler.Close();
+//                     return false;
+//                 }
+//
+//                 if (ToggleLevelMenuHandler.instance.mapMenuCanvas.activeInHierarchy)
+//                 {
+//                     ToggleLevelMenuHandler.instance.mapMenuCanvas.SetActive (false);
+//                     return false;
+//                 }
+//                 
+//                 if (!ToggleCardsMenuHandler.disableEscapeButton && ToggleCardsMenuHandler.cardMenuCanvas.activeInHierarchy)
+//                 {
+//                     ToggleCardsMenuHandler.SetActive(ToggleCardsMenuHandler.cardMenuCanvas.transform, false);
+//                     if(ToggleCardsMenuHandler.menuOpenFromOutside) ToggleCardsMenuHandler.Close();
+//                     return false;
+//                 }
+//             }
+//
+//             if (ModOptions.inPauseMenu)
+//             {
+//                 return false;
+//             }
+//             
+//
+//             return true;
+//         }
+//     }
+// }
```

Comment:
This has always done nothing as a patch, left the file so it can be rewritten in the future without forgetting

Reply:
Or at least has not worked with changes that have been made a couple of months ago

Reply:
It doesn't look like it's doing nothing? Anybody know what it was here in the first place, and what its intention is/was

Reply:
The intention was to block of from closing the settings menu when in the toggle cards/maps menu and pressing the escape key, though the patch doesn't nothing to prevent that as it always goes back to the main menu, this can be fixed but I have just commented it out for now as it always returned before doing any of that logic being always null